pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -66,15 +79,40 @@ def on_trainer_init(
         self.trainer.reload_dataloaders_every_n_epochs = reload_dataloaders_every_n_epochs
         self.trainer._is_data_prepared = False
 
-    def get_profiled_train_dataloader(self, train_dataloader) -> Iterable:
-        # FIXME: Temporary hack
-        if isinstance(self.data_fetcher, InterBatchParallelDataFetcher):
-            self.data_fetcher.setup(train_dataloader, batch_to_device=self.trainer.accelerator.batch_to_device)
+    def _check_training_step_requires_dataloader_iter(self) -> bool:
+        if not self.trainer.training:
+            return False
```

Comment:
This if should be placed where this function is called, not inside the function.

`if self.trainer.training and self._check_training_step_requires_dataloader_iter()`

Reply:
Done !